CoreFoundation/Locale.subproj/CFCalendar.c

```diff
@@ -5,29 +5,352 @@
 	Licensed under Apache License v2.0 with Runtime Library Exception
 	See http://swift.org/LICENSE.txt for license information
 	See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
-	Responsibility: Christopher Kane
 */
 
-
 #include <CoreFoundation/CFCalendar.h>
-#include <CoreFoundation/CFLocaleInternal.h>
 #include <CoreFoundation/CFRuntime.h>
 #include "CFInternal.h"
-#include "CFPriv.h"
 #include "CFRuntime_Internal.h"
-#include <unicode/ucal.h>
+#include "CFPriv.h"
+#include "CFCalendar_Internal.h"
+#include "CFLocaleInternal.h"
+#include "CFICULogging.h"
+#include "CFDateInterval.h"
+#include <assert.h>
 
-#define BUFFER_SIZE 512
 
-struct __CFCalendar {
-    CFRuntimeBase _base;
-    CFStringRef _identifier;	// canonical identifier, never NULL
-    CFLocaleRef _locale;
-    CFStringRef _localeID;
-    CFTimeZoneRef _tz;
-    UCalendar *_cal;
+// This avoids having to use a deprecated enum value throughout the implementation
+enum {
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated"
+    kCFCalendarUnitWeek_Deprecated = kCFCalendarUnitWeek,
+#pragma GCC diagnostic pop
 };
 
+
+#define ICU_LOG(FMT, ...) do { } while (0)
+
+extern CFDictionaryRef __CFLocaleGetPrefs(CFLocaleRef locale);
+
+#define MIN_TIMEZONE_UDATE -2177452800000.0  // 1901-01-01 00:00:00 +0000
+#define MAX_TIMEZONE_UDATE  4133980800000.0  // 2101-01-01 00:00:00 +0000
+
+CF_PRIVATE Boolean __calcNextDaylightSavingTimeTransition(UCalendar *ucal, UDate start_udate, UDate limit, UDate *answer) {
+    if (start_udate < MIN_TIMEZONE_UDATE) start_udate = MIN_TIMEZONE_UDATE; // answer question as if this date for anything earlier
+    if (MAX_TIMEZONE_UDATE < limit) limit = MAX_TIMEZONE_UDATE; // limit gets the smaller
```

Comment:
I’m going to take non-essential fixes as follow-ups, since the focus of this is to merge code verbatim and make sure it works as-is from our internal implementation first.